packages/pglite/src/utils.ts

```diff
@@ -5,30 +5,74 @@ export const IN_NODE =
   typeof process.versions === 'object' &&
   typeof process.versions.node === 'string'
 
-export async function makeLocateFile() {
-  const PGWASM_URL = new URL('../release/postgres.wasm', import.meta.url)
-  const PGSHARE_URL = new URL('../release/postgres.data', import.meta.url)
-  let fileURLToPath = (fileUrl: URL) => fileUrl.pathname
+let wasmDownloadPromise: Promise<Response> | undefined
+
+export async function startWasmDownload() {
   if (IN_NODE) {
-    fileURLToPath = (await import('url')).fileURLToPath
+    return
   }
-  return (base: string) => {
-    let url: URL | null = null
-    switch (base) {
-      case 'postgres.data':
-        url = PGSHARE_URL
-        break
-      case 'postgres.wasm':
-        url = PGWASM_URL
-        break
-      default:
-        console.error('makeLocateFile', base)
-    }
+  const moduleUrl = new URL('../release/postgres.wasm', import.meta.url)
+  wasmDownloadPromise = fetch(moduleUrl)
```

Comment:
nit: guard against multiple calls here?

(also, I'm trembling at potential bugs here haha but I don't see why this wouldn't work - did CORS or other headers get added to bundle requests before?)

Reply:
:man_facepalming: I meant to add that....

A thought, should we also add a `fetchClient` similar to what we do in the Electric `ShapeStream` so that people can configure CORS headers?

Reply:
eh, it feels like it errs on the side of overcustomization but I suppose this PR does address very particular resource acquisition cases, so might be a good idea?